Purchases/Purchasing/OfferingsManager.swift

```diff
@@ -160,7 +158,7 @@ private extension OfferingsManager {
               }
 
         let productIdentifiers = extractProductIdentifiers(fromOfferingsData: offeringsData)
-        let missingProducts = Set(products.keys).intersection(productIdentifiers)
+        let missingProducts = productIdentifiers.subtracting(Set(products.keys))
```

Comment:
it looks like some of [this](https://github.com/RevenueCat/purchases-ios/pull/868/files) code got overwritten at some point, and the function added there isn't used anymore. could we remove it in this PR?

Reply:
Oh thanks! Funny that we both solved the same issue, but I like your implementation better.
Looks like this was probably a bad merge in #953. I'll update your PR.

Reply:
Updated.